pages/api/level/[id]/[slugName].ts

```diff
@@ -0,0 +1,45 @@
+import withAuth, { NextApiRequestWithAuth } from '../../../../lib/withAuth';
+import Level from '../../../../models/db/level';
+import { LevelModel } from '../../../../models/mongoose';
+import type { NextApiResponse } from 'next';
+import dbConnect from '../../../../lib/dbConnect';
+
+export default withAuth(async (req: NextApiRequestWithAuth, res: NextApiResponse) => {
+  if (req.method !== 'GET') {
+    return res.status(405).json({
+      error: 'Method not allowed',
+    });
+  }
+
+  await dbConnect();
+
+  const {params} = req.query;
+  const level = await getLevelByUrlPath(params as string[]);
+  if (!level) {
+    return res.status(404).json({
+      error: 'Level not found',
+    });
+  }
+  return res.status(200).json(level);
+});
+
+export async function getLevelByUrlPath(params: string[]):Promise<Level | null> {
+  let level;
+  if (params.length === 1) {
+    const id = params[0];
```

Comment:
If one parameter is passed, wouldn't it match the existing `api/level/[id].ts`?

Reply:
Actually... Just realized (after removing it). The fallback [id] that it would match works for api, but not for pages. So with this PR currently it would mean that access a level by ID wouldn't render.

What do you think? Should we remove the ability to access a level page by its id?